gtests/net/packetdrill/run_packet.c

```diff
@@ -371,6 +371,31 @@ static struct socket *find_connect_for_live_packet(
 	return socket;
 }
 
+static struct socket *find_mp_join_for_live_packet(
+	struct state *state, struct packet *packet,
+	enum direction_t *direction)
+{
+	struct tuple tuple;
+	struct socket *socket = state->socket_under_test;	/* shortcut */
+	if (!socket)
+		return NULL;
+
+	*direction = DIRECTION_INVALID;
+	get_packet_tuple(packet, &tuple);
+
+         if (packet->tcp && packet->tcp->syn && !packet->tcp->ack &&
+	     socket->protocol == IPPROTO_MPTCP &&
+	     socket->state == SOCKET_ACTIVE_MP_JOIN_SENT) {
```

Comment:
no need to check if the package is linked to an existing MPTCP connection?

Reply:
actually this is a design flaw in current packetdrill. The current code first tries to match the outbound packet with a socket, then it validates the packet contents (including TCP options). So, this "check" must done later (i.e. by validating the value of the token).

Reply:
Yes, I suspected that, thank you for the clarification!

And this check is done, right?